plugins/modules/cm_autotls.py

```diff
@@ -0,0 +1,462 @@
+# Copyright 2025 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible_collections.cloudera.cluster.plugins.module_utils.cm_utils import (
+    ClouderaManagerModule,
+)
+from cm_client.rest import ApiException
+from cm_client import (
+    ClouderaManagerResourceApi,
+    ApiGenerateCmcaArguments,
+    ApiConfigList,
+    ApiConfig,
+)
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: cm_autotls
+short_description: Manage and configure Auto-TLS and Cloudera Manager CA
+description:
+  - Enables and configures Auto-TLS and Cloudera Manager as a CA.
+  - Disabling of Auto-TLS is also supported.
+  - Note that disabling Auto-TLS does not remove the TLS resources (keys, truststores, etc.) created during the enable process.
+author:
+  - "Jim Enright (@jimright)"
+requirements:
+  - cm_client
+options:
+  state:
+    description:
+      - The declarative state of Auto-TLS.
+      - Disabling Auto-TLS does not remove the TLS resources (keys, truststores, etc.) created during the enable process.
+    type: str
+    required: false
+    default: present
+    choices:
+      - present
+      - absent
+  force:
+    description:
+      - Forces enabling Auto-TLS even if it is already determined to be enabled.
+      - Applicable only when I(state) is C(true).
```

Comment:
Update to the current best practices for markup: https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#semantic-markup-within-module-documentation

Reply:
Done